typing/outcometree.mli

```diff
@@ -126,7 +133,7 @@ and out_type_decl =
     otype_params: out_type_param list;
     otype_type: out_type;
     otype_private: Asttypes.private_flag;
-    otype_immediate: Type_immediacy.t;
+    otype_layout: out_layout_annot option;
```

Comment:
At a glance, this doesn't seem equivalent to the old bit of information here. This new piece sounds like it's about surface syntax, whereas the old one sounds like it's about something computed during type-checking. But I could be wrong, as I didn't look deeply.